app/Jobs/DispatchPollJobs.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Models\Device;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Log;
+use LibreNMS\Config;
+
+class DispatchPollJobs implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * @param  string  $device_spec
+     * @param  int  $verbosity
+     * @param  array<string, bool|string[]>  $module_overrides
+     */
+    public function __construct(
+        public string $device_spec,
+        public int $verbosity = -1,
+        public array $module_overrides = [],
+    ) {
+    }
+
+    /**
+     * Execute the job.
+     */
+    public function handle()
+    {
+        // Make sure we have configured job queueing
+        if (\config('queue.default') != 'sync') {
```

Comment:
This should be blocked from running on every current install until the user intervenes.  The current code does not meet that standard.

I do have positive thoughts about a config setting that explicitly sets the controller type. That should maybe be a separate PR as if we want to apply it to existing polling methods (of which there are 3, this is the 4th).

One downside there, is it seems like there is a lot of opportunity for users to break polling. For example, config settings can be overridden in local config.php files.  So, users could have different settings on different nodes for what their polling method is.

Note: I prefer the early return pattern over nested ifs

Reply:
This is a section of code that I have also raised questions about.  I wasn't aware that there was any use case for having job queues.  I would prefer to wait for a good solution to the config issue instead of adding another setting to clean up later.

I hadn't considered the use of config.php overrides, or having different polling methods on different nodes.

How does this sound for an idea on the config option:
 - Have a "legacy" option that uses any existing defined config options. This should be the default for existing installs.
 - Have options for each of the supported methods
 - Update each polling method to only run if one of the following is true:
  - The new config option is set to "legacy" and any existing conditions are met (e.g. for the dispatch worker service_poller_enabled is also set)
  - The new config option is set to the correct polling method

Going forward, we would want users to override the new option if they want to override the polling method on a server.

I also think that we should create options for all the cron jobs (normal discovery, new device discovery, alerts, billing, services and fast ping) while we're at it to make overriding them easier in future.

Reply:
I agree on all points.

I'm leaving out the other processes for now to keep things simple.  But for the config setting, do you think it would make sense to allow them to be set separately, or should it be all one big setting?

We could probably add some new validation checks too after we figure out how people will mess it up :)